### ISSUE TITLE
 Grafana Tempo  shows too many references

### ISSUE THREAD

**Issue by dkowalewski** [grafana/tempo#3869]
Grafana Tempo  shows too many references.

What I'm doing:
1. Create Trace1 with Span1.
2. Create Trace2 with Span2 linked to Span1.

For Span2, Grafana Tempo shows two references:
- To Trace2 with 000000000 as the SpanId - it looks like a reference to itself;
- To Trace1 with Span1 - as expected;

![Screenshot from 2024-07-16 12-06-43](https://github.com/user-attachments/assets/61b0887b-3a5a-484d-9f77-1b2c24d56ec7)

Logs from Collector shows that exporter sended span with correct one link:
![Screenshot from 2024-07-16 12-08-14](https://github.com/user-attachments/assets/bbbfd017-4df8-4a0b-9676-c3924a879522)

Export from tempo also looks good:
[Trace-edcd68-2024-07-16 12_13_31.json](https://github.com/user-attachments/files/16247094/Trace-edcd68-2024-07-16.12_13_31.json)

I found issue  https://github.com/grafana/tempo/issues/3534  which I think describes the same issue. Turning on debugging received spans does not give me any clue (links are not logged).
Why tempo shows first not needed reference ? What I'm doing wrong ?

**Environment:**
Collector, Grafan, Tempo  deployed on Ubuntu machine with docker-compose.
Spans created with simple Java App.

**Comment by joe-elliott** [comment]
Can you please curl the `/api/traces/<traceid>` endpoint and see if Tempo is returning this data? If Tempo is returning it we will look into it on our side. If not this is a rendering issue and we will move to Grafana.

Thanks!

**Comment by dkowalewski** [comment]
Yes here it is:

![image](https://github.com/user-attachments/assets/b6e1f1a3-d1d7-4a2a-929b-6a6552ceba92)
[2344d37140214cd8eb693c8f9e745afb.json](https://github.com/user-attachments/files/16249752/2344d37140214cd8eb693c8f9e745afb.json)

![image](https://github.com/user-attachments/assets/a58b0439-80a8-4654-aa24-4c7374e2484c)

**Comment by joe-elliott** [comment]
it seems like there's only one link in the json? and its the second one in the screenshot?

```
$ echo qfqZinjTkJo= | base64 -d | xxd
00000000: a9fa 998a 78d3 909a                      ....x...

$ echo uYeZ8+SAFpgzfhamm+BXuA== | base64 -d | xxd
00000000: b987 99f3 e480 1698 337e 16a6 9be0 57b8  ........3~....W.
```

**Comment by dkowalewski** [comment]
Yes.  In the json there is only one link. Grafana shows two links. It's seems to be rendering problem.
